libclamav/crtmgr.c

```diff
@@ -303,132 +450,100 @@ static int crtmgr_rsa_verify(cli_crt *x509, fp_int *sig, cli_crt_hashtype hashty
         return 1;
     }
 
-    fp_init(&x);
+    if (MAX(keylen, siglen) - MIN(keylen, siglen) > 1) {
+        cli_dbgmsg("crtmgr_rsa_verify: keylen and siglen differ by more than one\n");
+        return 1;
+    }
 
-    do {
-        if (MAX(keylen, siglen) - MIN(keylen, siglen) > 1) {
-            cli_dbgmsg("crtmgr_rsa_verify: keylen and siglen differ by more than one\n");
-            break;
-        }
-        if ((ret = fp_exptmod(sig, &x509->e, &x509->n, &x))) {
-            cli_warnmsg("crtmgr_rsa_verify: verification failed: fp_exptmod failed with %d\n", ret);
-            break;
-        }
-        if (fp_unsigned_bin_size(&x) != keylen - 1) {
-            cli_dbgmsg("crtmgr_rsa_verify: keylen-1 doesn't match expected size of exptmod result\n");
-            break;
-        }
-        if (((unsigned int)fp_unsigned_bin_size(&x)) > sizeof(d)) {
-            cli_dbgmsg("crtmgr_rsa_verify: exptmod result would overrun working buffer\n");
-            break;
-        }
+    ret = crtmgr_get_recorv_data(sig, x509, &buff, &d, &len);
```

Comment:
I haven't quite followed the logic changes to understand why some of the more extensive changes here (not complaining! just a little overwhelmed).   I wonder if you could elaborate a little to help me review.

Reply:
Thank you Sebastian!  That helps